### DISCUSSION TITLE
Rerender Table using the Angular way

### DISCUSSION THREAD

**Discussion by Mafiosso5** [l-lin/angular-datatables#1619] [Q&A]
How do i do the Rerender functionality whilst using the Angular way of getting data in the table? This is for showing new data in the table

**Comment by venukalyanvella** [comment]
datatables was not updating on angulat 12 and i have a query is the datatables 13.0.0 was supports angular 12 or not ?

**Comment by shanmukhateja** [comment]
Hi @Mafiosso5 

I don't know the exact problem without knowing how you've setup the app.

If this is regarding fetching new data after user action is performed, move the HTTP request code to a separate function and call it as needed.

 For a more accurate answer, please provide a reproducible StackBlitz or GitHub repo.

**Reply by Mafiosso5** [reply]
Hie @shanmukhateja 

Here is the repository : https://github.com/Mafiosso5/angular-datatable.git

I want to rerender the data table after I filter the data using the date range filter. When i try the table does not render properly.

**Reply by shanmukhateja** [reply]
Hi @Mafiosso5 could you replace `this.dtElement.dtInstance.then` with `this.dtElement.dtInstance?.then` in `dashboard.component.ts`?

After adding optional chaining operator, it worked as expected. If you checked your console logs, you'll see after the date is selected and Apply is clicked (in the modal), an error is thrown about accessing `then` of undefined.

**Reply by Mafiosso5** [reply]
Hie @shanmukhateja 

I have made the change you have suggested, but nothing has changed. I is still not rerendering properly.

I have updated the repository to reflect the change you suggested for you to see or yourself

**Reply by shanmukhateja** [reply]
Hi @Mafiosso5 

I tried setting date range from 01-June-2021 to 31-June-2021 and it seems to be working as expected.

**Before filter: **
![ss01](https://user-images.githubusercontent.com/43369891/148011083-34e36a5a-6c73-4fa6-83b5-da78d9f56378.png)

**After filter:**
![ss02](https://user-images.githubusercontent.com/43369891/148011099-fd939328-cb65-4206-9105-822066d17a19.png)

Could you explain "what" is supposed to work but it isn't now?

**Reply by Mafiosso5** [reply]
Hie @shanmukhateja 

The table in the 2nd screenshot after the filter has no search bar, no pagination, ordering or CSS.

It does not look the same as the 1st screenshot.

I want to maintain the same attributes of data-tables even after a filter like the example set in the documentation

**Reply by shanmukhateja** [reply]
Could you apply the changes as mentioned here?
 https://pastebin.com/2mt5PNKJ

**Reply by thanjeys** [reply]
Hi @shanmukhateja 

What is solution for this problem. I am also stucking same issue ? 

Thanks
